Cloudflare Pages adapter is incompatible with newer workerd's `PagesFunction` type

With the following `tsconfig.json` and `index.ts`, TypeScript produces a compile error.

```json5
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "lib": ["esnext"],
    "moduleResolution": "node",
    // note the compatibility date (I need to set this in order to get the correct type for FormData).
    // if I omit the date, the error disappears.
    "types": ["@cloudflare/workers-types/2022-08-04"],
    "noEmit": true
  }
}
```

```typescript
import { Hono } from "hono";
import { handle } from "hono/cloudflare-pages";

const app = new Hono();
app.get("/hello", (c) => c.text("Hello from Hono!"));

export const onRequest: PagesFunction = handle(app);
```

The error message is:

```console
index.ts:7:14 - error TS2322: Type '(eventContext: EventContext<{}, string, {}>) => Response | Promise<Response>' is not assignable to type 'PagesFunction<unknown, any, Record<string, unknown>>'.
  Types of parameters 'eventContext' and 'context' are incompatible.
    Type 'EventContext<unknown, any, Record<string, unknown>>' is not assignable to type 'EventContext<{}, string, {}>'.
      The types of 'request.fetcher.fetch' are incompatible between these types.
        Type '(input: RequestInfo, init?: RequestInit<RequestInitCfProperties>) => Promise<Response>' is not assignable to type '(input: import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInfo, init?: import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInit<import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInitCfProperties>) => Pr...'.
          Types of parameters 'input' and 'input' are incompatible.
            Type 'import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInfo' is not assignable to type 'RequestInfo'.
              Type 'Request<unknown>' is not assignable to type 'RequestInfo'.
                Type 'import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").Request<unknown>' is not assignable to type 'Request<unknown>'.
                  The types of 'fetcher.fetch' are incompatible between these types.
                    Type '(input: import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInfo, init?: import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInit<import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInitCfProperties>) => Pr...' is not assignable to type '(input: RequestInfo, init?: RequestInit<RequestInitCfProperties>) => Promise<Response>'.
                      Types of parameters 'init' and 'init' are incompatible.
                        Type 'RequestInit<RequestInitCfProperties>' is not assignable to type 'import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInit<import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").RequestInitCfProperties>'.
                          Types of property 'body' are incompatible.
                            Type 'BodyInit' is not assignable to type 'import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").BodyInit'.
                              Type 'FormData' is not assignable to type 'BodyInit'.
                                Type 'FormData' is not assignable to type 'import("/Users/ynkt/tmp/repro-hono/node_modules/@cloudflare/workers-types/index").FormData'.
                                  The types returned by 'get(...)' are incompatible between these types.
                                    Type 'string | File' is not assignable to type 'string'.
                                      Type 'File' is not assignable to type 'string'.

7 export const onRequest: PagesFunction = handle(app);
               ~~~~~~~~~
```
I don't have time to investigate further right now (time to go to bed here lol), but will look into this tomorrow's night.

--- a/src/adapter/cloudflare-pages/handler.ts
+++ b/src/adapter/cloudflare-pages/handler.ts
@@ -1,11 +1,16 @@
 // @denoify-ignore
-/// <reference types="@cloudflare/workers-types" />
 import type { Hono } from '../../hono'
 import type { Env } from '../../types'
 
+type EventContext = {
+  request: Request
+  waitUntil: (promise: Promise<unknown>) => void
+  env: {} & { ASSETS: { fetch: typeof fetch } }
+}
+
 export const handle = <E extends Env>(app: Hono<E>) => {
-  return (eventContext: EventContext<{}, string, {}>): Response | Promise<Response> => {
-    const { request, env, waitUntil, passThroughOnException } = eventContext
-    return app.fetch(request, env, { waitUntil, passThroughOnException })
+  return (eventContext: EventContext): Response | Promise<Response> => {
+    const { request, env, waitUntil } = eventContext
+    return app.fetch(request, env, { waitUntil, passThroughOnException: () => {} })
   }
 }

--- a/src/adapter/cloudflare-pages/handler.ts
+++ b/src/adapter/cloudflare-pages/handler.ts
@@ -1,13 +1,17 @@
 // @denoify-ignore
-import type { EventContext } from '@cloudflare/workers-types'
 import type { Hono } from '../../hono'
 import type { Env } from '../../types'
 
+type EventContext = {
+  request: Request
+  waitUntil: (promise: Promise<unknown>) => void
+  passThroughOnException: () => void
+  env: {} & { ASSETS: { fetch: typeof fetch } }
+}
+
 export const handle = <E extends Env>(app: Hono<E>) => {
-  return (eventContext: EventContext<{}, string, {}>) => {
+  return (eventContext: EventContext): Response | Promise<Response> => {
     const { request, env, waitUntil, passThroughOnException } = eventContext
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     return app.fetch(request, env, { waitUntil, passThroughOnException })
   }
 }

--- a/src/adapter/cloudflare-pages/handler.ts
+++ b/src/adapter/cloudflare-pages/handler.ts
@@ -1,16 +1,10 @@
 // @denoify-ignore
+/// <reference types="@cloudflare/workers-types" />
 import type { Hono } from '../../hono'
 import type { Env } from '../../types'
 
-type EventContext = {
-  request: Request
-  waitUntil: (promise: Promise<unknown>) => void
-  passThroughOnException: () => void
-  env: {} & { ASSETS: { fetch: typeof fetch } }
-}
-
 export const handle = <E extends Env>(app: Hono<E>) => {
-  return (eventContext: EventContext): Response | Promise<Response> => {
+  return (eventContext: EventContext<{}, string, {}>): Response | Promise<Response> => {
     const { request, env, waitUntil, passThroughOnException } = eventContext
     return app.fetch(request, env, { waitUntil, passThroughOnException })
   }

